tests/unit/resources/test_compensation.py

```diff
@@ -98,3 +108,143 @@ def test_create_from_spill_string(spillstring):
         "Qdot655-A",
         "Qdot705-A",
     ]
+
+
+@responses.activate
+def test_apply_scale_errors_for_nonmatching_channels(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = parse_fcs_file(client.download_fcs_file(EXP_ID, fcs_file._id))
+    fcs_file.events = events
+
+    with pytest.raises(IndexError):
+        compensation.apply(fcs_file)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file(compensation, fcs_file):
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    events = parse_fcs_file(events_body.read())
+    fcs_file.events = events
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24), index=ix, columns=ix)
+    compensation.channels = ix
+
+    results = compensation.apply(fcs_file)
+
+    assert all(results == events)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file_with_no_events(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = fcs_file.events
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24), index=ix, columns=ix)
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file)
+
+    # Then: events should be scaled
+    assert all(results == events)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file_with_matching_kwargs(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=open("tests/data/Acea - Novocyte.fcs", "rb"),
+    )
+    events = fcs_file.get_events(inplace=True, testKwarg="foo")
+    assert fcs_file._events_kwargs == {"testKwarg": "foo"}
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24), index=ix, columns=ix)
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file, testKwarg="foo")
+
+    # Then: events should be scaled
+    assert all(results == events)
+    assert (
+        responses.assert_call_count(
+            f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs?testKwarg=foo",
+            1,
+        )
+        is True
+    )
+
+
+@responses.activate
+def test_apply_scale_with_subset_of_values(
```

Comment:
If you improve the assertion in `test_apply_scale_compensates_values`, then you can delete this test also. The file-internal comp in the Novocyte file has 18 of the file's 24 channels.